### PR TITLE
pending deprecations from 0.21

### PR DESCRIPTION
**Pull request by 1ucian0**
There are some pending deprecations that can be deprecations in 1.1 (to be removed in 2.0).

```
0.21: Released in June 30, 2022 (wrt last minor release, 19 month old)
 - qiskit/transpiler/passes/scheduling/alap.py:30 (__init__) - PENDING
 - qiskit/transpiler/passes/scheduling/asap.py:36 (__init__) - PENDING
 - qiskit/transpiler/passes/scheduling/dynamical_decoupling.py:107 (__init__) - PENDING
 - qiskit/transpiler/passes/scheduling/alignments/align_measures.py:93 (__init__) - PENDING
```

The deprecated classes are:
 * `ALAPSchedule`
 * `ASAPSchedule`
 * `DynamicalDecoupling`
 * `AlignMeasures`

### CODE REVIEW

**Comment by jakelishman** [comment]
Aside from the test failures here, I'm also relatively sure that these classes are used in some documentation examples - I think I've looked at doing these deprecations before.

**Comment by 1ucian0** [comment]
Thanks! Tests fixed.

I also searched for the deprecated classes [in `documentation/docs/transpile`](https://github.com/Qiskit/documentation/tree/main/docs/transpile) and I got no hits. I can raise an issue in [Qiskit/documentation](https://github.com/Qiskit/documentation) if you feel strongly that something can break there.

File: releasenotes/notes/removed_deprecated_0.21-741d08a01a7ed527.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+deprecations:
```
Comment (jakelishman) [inline releasenotes/notes/removed_deprecated_0.21-741d08a01a7ed527.yaml]:
```suggestion
deprecations_transpiler:
```

**Review by jakelishman — COMMENTED** [review]
Ok, if you've got all the places in the documentation (and from my quick search, it does just look like they're used in the docstring of `DynamicalDecoupling`, which is itself deprecated, so no worries), then I think this is fine.  Since #11819 is on `main`, we can start categorising our release notes during merge (so there's less to do later), which should hopefully help out releases.

**Review by jakelishman — APPROVED** [review]
I've forgotten the context of this a little bit now, but it seems about right, and I'd not back myself to know all the nuances of the legacy scheduling passes anyway.

**Pull request merged**